### DISCUSSION TITLE
Move from github?

### DISCUSSION THREAD

**Discussion by yaxu** [tidalcycles/Tidal#1121] [General]
Some have deleted their github accounts for ethical reasons and so are excluded from discussions here.

Gitea looks like it's well maintained and with a familiar interface. I can't see any big disadvantages to moving to that?

**Comment by telephon** [comment]
Looks interesting! Is it based on Gitlab?

**Comment by yaxu** [comment]
No I think its a separate code base, and lighter on resources. Migration of issues etc looks straightforward

**Comment by ndr-brt** [comment]
Ok for me.
we shoud learn and adopt another continuous environment system. Don't know what integrates well with gitea

**Comment by yaxu** [comment]
It looks like there are a couple of options:
  https://gitea.com/gitea/awesome-gitea#devops

**Comment by ndr-brt** [comment]
Nice. In the past I used jenkins a lot, and I also explored concourse.
Seems like `agola`, `drone` and `woodpecker` have a built-in support for gitea.

**Comment by yaxu** [comment]
I've put a mirror up on codeberg, which runs gitea: https://codeberg.org/TidalCycles

It seems nice!

If we decide to move there, there's a migration function to bring issues etc across. 

*edit* I removed the mirror, I thought it would be confusing to have both active. If/when we move, we would have needed to delete that anyway and use the migration tool.

**Comment by yaxu** [comment]
I've tried migrating the vortex project there: https://codeberg.org/TidalCycles/vortex

It only had one issue but seems smooth..

**Comment by yaxu** [comment]
We moved vortex back to github, as codeberg didn't seem quite there in terms of running ci etc.

But perhaps we could try gitlab? At the same time, we could change the organisational structure a bit to reflect the appearance of strudel and vortex. I propose we have a top-level group 'algopattern' with 'tidalcycles', 'strudel' and 'vortex' as subgroups of that.

There is an import process, but before running it we'd want to have as many people with gitlab accounts as we can so that issues and pull/merge requests are properly assigned.

**Comment by yaxu** [comment]
I imported feedforward there as a test https://gitlab.com/algopattern/tidalcycles/feedforward

**Comment by sss-create** [comment]
Using codeberg would resonate with me. Otherwise this may be closed?

**Reply by yaxu** [reply]
> Using codeberg would resonate with me. Otherwise this may be closed?

I thought best to convert it to a discussion

**Reply by cleary** [reply]
It feels like the appetite to move from github has waned a bit since the Microsoft buyout - most tidal and periphery projects I'm aware of are still using github.

I've been running self hosted gogs at work for a few years - it's not suitable for this, I'm migrating to gitea in coming weeks (gogs fork), but I still don't think it's quite there with ci options

Codeberg I'm unfamiliar with, another option that the fedora project moved to recently is forgejo - I have no other info about it though

**Reply by sss-create** [reply]
Codeberg uses Forgejo and I think Woodpecker for CI.

**Reply by yaxu** [reply]
I think more recent political events should bring this back on the agenda. 

I love the codeberg ethos, but my impression from earlier experience is that gitlab would be easier in terms of actions etc?

**Reply by sss-create** [reply]
fwiw I read that GitLab explores sale. This was a few months ago but you never know. I am not sure about actions, but I suppose our pipelines are transferable. I did some CI tests on sourcehut recently when the Nix issues came up and it was easy to migrate.
